pkgs/common-updater/scripts/update-source-version

```diff
@@ -229,10 +229,11 @@ if [[ -n "$newUrl" ]]; then
         die "Couldn't evaluate source url from '$attr.$sourceKey'!"
     fi
 
-    # Escape regex metacharacter that are allowed in store path names
-    oldUrlEscaped=$(echo "$oldUrl" | sed -re 's|[${}.+?]|\\&|g')
+    # Escape regex metacharacter that may appear in URLs
+    oldUrlEscaped=$(echo "$oldUrl" | sed -re 's|[${}.+?\[]|\\&|g')
+    newUrlEscaped=$(echo "$newUrl" | sed -re 's|&|\\&|g')
```

Comment:
I believe these are the full set of characters that need escaping with [BRE syntax](https://www.gnu.org/software/sed/manual/html_node/BRE-syntax.html#BRE-syntax) (`|` because it is used as separator):

```suggestion
    oldUrlEscaped=$(echo "$oldUrl" | sed -e 's|[*.^$[\|]|\\&|g')
    newUrlEscaped=$(echo "$newUrl" | sed -e 's|[&\|]|\\&|g')
```